lib/componentRelay.ts

```diff
@@ -0,0 +1,754 @@
+    this.postMessage(ComponentAction.DeassociateItem, { item: this.jsonObjectForItem(item) } )
+  }
+
+  /**
+   * Deletes an Item from the item store.
+   * @param item The Item to delete.
+   * @param callback The callback with the result of the operation.
+   */
+  public deleteItem(item: SNItem, callback: (data: any) => void) : void {
+    this.deleteItems([item], callback)
+  }
+
+  /**
+   * Deletes a collection of Items from the item store.
+   * @param items The Item(s) to delete.
+   * @param callback The callback with the result of the operation.
+   */
+  public deleteItems(items: SNItem[], callback: (data: any) => void) : void {
+    const params = {
+      items: items.map((item: SNItem) => {
+        return this.jsonObjectForItem(item)
+      }),
+    }
+    this.postMessage(ComponentAction.DeleteItems, params, (data) => {
+      callback && callback(data)
+    })
+  }
+
+  /**
+   * Performs a custom action to the component manager.
+   * @param action
+   * @param data 
+   * @param callback The callback with the result of the operation.
+   */
+  public sendCustomEvent(action: ComponentAction, data: any, callback?: (data: any) => void) : void {
+    this.postMessage(action, data, (data: any) => {
+      callback && callback(data)
+    })
+  }
+
+  /**
+   * Saves an existing Item in the item store.
+   * @param item An existing Item to be saved.
+   * @param callback 
+   * @param skipDebouncer 
+   */
+  public saveItem(item: SNItem, callback?: () => void, skipDebouncer = false) : void {
+    this.saveItems([item], callback, skipDebouncer)
+  }
+
+  /**
+   * Runs a callback before saving an Item.
+   * @param item An existing Item to be saved.
+   * @param presave Allows clients to perform any actions last second before the save actually occurs (like setting previews).
+   * Saves debounce by default, so if a client needs to compute a property on an item before saving, it's best to
+   * hook into the debounce cycle so that clients don't have to implement their own debouncing.
+   * @param callback
+   */
+  public saveItemWithPresave(item: SNItem, presave: any, callback: () => void) : void {
+    this.saveItemsWithPresave([item], presave, callback)
+  }
+
+  /**
+   * Runs a callback before saving a collection of Items.
+   * @param items A collection of existing Items to be saved.
+   * @param presave Allows clients to perform any actions last second before the save actually occurs (like setting previews).
+   * Saves debounce by default, so if a client needs to compute a property on an item before saving, it's best to
+   * hook into the debounce cycle so that clients don't have to implement their own debouncing.
+   * @param callback
+   */
+  public saveItemsWithPresave(items: SNItem[], presave: any, callback: () => void) : void {
+    this.saveItems(items, callback, false, presave)
+  }
+
+  private _performSavingOfItems({ items, presave, callback }: { items: SNItem[], presave: () => void, callback?: () => void }) {
```

Comment:
Function signature doesn't need underscore now that we're using TS with `private` modifier.